pkg/agent/cniserver/ipam/annotations.go

```diff
@@ -15,6 +15,9 @@
 package ipam
 
 const (
-	AntreaIPAMAnnotationKey       = "ipam.antrea.io/ippools"
-	AntreaIPAMAnnotationDelimiter = ","
+	AntreaIPAMAnnotationKey            = "ipam.antrea.io/ippools"
+	AntreaIPAMPodIPAnnotationKey       = "ipam.antrea.io/pod-ips"
+	AntreaIPAMPersistIPAnnotationKey   = "ipam.antrea.io/persist-ips"
```

Comment:
I suggest to define this as global flag to keep it simple

Reply:
Not get your point. Could you give an example?

Reply:
I think we decided we always persist StatefulSet IPs? We can have a global flag as @annakhm suggests, but I feel that is not important in my mind.

Reply:
I think an annotation is more flexible, since a global flag requires a restart of antrea-agent, and the "global reserve" feature will waste many IPs if some StatefulSets don't need it.

Reply:
Why it wastes IPs? No matter we persists IP or not, StatefulSet always requires the same number of IPs. Persistence just guarantees the same Pod gets the same IP after restarts, which has no harm at all. And I think no need to ask one more annotation to ask for persistence.

Reply:
With current implementation, we can decrease the StatefulSet replica size and then increase the replica size, and agent will use reserved IPs first. (persistent IP for scaling StatefulSet down and up) The reserved IPs will be kept until StatefulSet is deleted.
Do you mean we only need to keep StatefulSet IP persistent when StatefulSet restart? (persistent IP for StatefulSet restart)

Reply:
I feel we should adjust reserved IPs based on replica size, but we can support that later. Changing replica size is not common anyway.

Reply:
How about this PR?
If we only support persistent IP for FlexibleIPAM StatefulSet restart in this PR, we can make this feature "always on".
Workflow change:
antrea-controller will handle StatefulSet update event to release reserved IPs if the index >= replicas.

Reply:
Yes, that is what I propose.

Reply:
@jianjuns OK. I'll remove this "persist-ip" annotation and make StatefulSet IP persistence "always on".
@annakhm Could you also add this into antrea-controller?
antrea-controller will handle StatefulSet update event to release reserved IPs if the index >= replicas.